### PR TITLE
Serve images locally if they don't get transferred.

### PR DESCRIPTION
**Pull request by pellaea**
In particular, this fixes the rails side.  There's an independent fix on the nginx side.

Here, we just need to make sure the entire `Image` instance is passed all the way in to the point where we calculate the image URL.  At least whenever possible.  There are still some cases (e.g. in textile markup such as `!image 1234!` where we do not have access to the instance and we just fall back on the default source.  But in the standard cases like the activity log and show obs page, we've eager-loaded the images already, so there is no reason not to pass it all the way in.

### CODE REVIEW

**Comment by pellaea** [comment]
I'm adding the (tiny) change to the nginx configuration to this PR, too.  It's just one of those strange nginx order things.  Even though the `rewrite` section appears at the end of the file, it still fires before the `try_files` business, because `rewrite` always fires before `try_files` regardless where they occur in the configuration file.  The solution is to shove the order-critical things in `location` blocks, because _those_ get executed in the order they appear.  Nice and easy, ya just gotta know.

**Comment by nimmolo** [comment]
Can we add a comment on the helpers to remind people the reason for these instance/id fallbacks?

**Comment by pellaea** [comment]
Sorry, I haven't written a unit test for this.  It will require temporarily changing `MO.image_precedence` for the space of at least one or two tests.  We need to have it try `:remote1` first then fallback on `:local`, not the other way around.  Then we can set `transferred` to false for some image that is present locally, then make sure the `image.url` for that image is local not remote.  Otherwise, if it prioritizes `:local`, then of course it will serve the local image, that won't test whether it's testing the `transferred` bit correctly on the remote image.  And then, even more important, because I think the `Image::URL` stuff is sufficiently tested already, we need to make sure the URLs that are rendered on the Activity Log and Show Obs pages are correct, too.  Phew.  Beyond the extremely limited scope of time I have available to me.  I tested this in development and it works fine.  So I'm just letting it go for now.  Besides the nginx configuration should catch anything we miss in rails, anyways.  We had a problem this time just because *both* failsafes were failing.

**Comment by pellaea** [comment]
Absolutely, comment away!

File: app/helpers/image_helper.rb [hunk 1]
```diff
@@ -105,12 +105,22 @@ def show_best_image(obs)
                       votes: true) + image_copyright(obs.thumb_image, obs)
   end
 
-  def original_image_link(image_id, classes)
-    link_to(:image_show_original.t, Image.url(:original, image_id),
+  def original_image_link(image_or_image_id, classes)
```
Comment (nimmolo) [inline app/helpers/image_helper.rb:108]:
```suggestion
  # pass an image instance if possible, to ensure access to fallback URL
  def original_image_link(image_or_image_id, classes)
```

File: app/helpers/lightbox_helper.rb [hunk 2]
```diff
@@ -43,11 +44,11 @@ def caption_obs_title(obs_data)
   end
 
   # links relating to the image object, pre-joined as a div
```
Comment (nimmolo) [inline app/helpers/lightbox_helper.rb:46]:
```suggestion
  # links relating to the image object, pre-joined as a div
  # pass an image instance if possible, to ensure access to fallback image.url
```

**Comment by nimmolo** [comment]
Comments added

**Pull request merged**